### PR TITLE
Add VIP: VET Custodian Contract Standard

### PR DESCRIPTION
**Pull request by laalaguer**
This VIP proposes a standard library, `StakingModel.sol` to better assist smart contract development by offloading VTHO-related calculations into a separate smart contract. It aims to ease the logic burden and prevent deadlock of VTHO in smart contracts on VeChain.

### CODE REVIEW

**Comment by NecoSherry** [comment]
### File `vips/VIP-VET-Custodian.md`

Required 3 approvals from @zzGHzz, @qianbin, @libotony, @mongelly, @asbertMa, @bacco1977, @nwbrettski, @vineet-codes, @BenMVeChain, @daithihearn

**Review by zzGHzz — APPROVED** [review]

**Review by wayne-qian — APPROVED** [review]

**Review by qianbin — APPROVED** [review]

**Review by sebaseek — CHANGES_REQUESTED** [review]
Hi, there is a typo in StakingModel.sol error definition

   ```
 function removeVTHO(address addr, uint256 amount) restrict(amount) internal {
        _update(addr);
        require(users[addr].energy >= uint104(amount), "insuffcient vtho");
        users[addr].energy -= uint104(amount);
    }

    function removeVET(address addr, uint256 amount) restrict(amount) internal {
        _update(addr);
        require(users[addr].balance >= uint104(amount), "insuffcient vet");
        users[addr].balance -= uint104(amount);
    }
```

" insuffcient " => "insufficient"

**Review by darrenvechain — APPROVED** [review]

File: vips/VIP-VET-Custodian.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
+---
+Title: VET Custodian Contract Standard
```
Comment (NecoSherry) [inline vips/VIP-VET-Custodian.md:2]:
```suggestion
VIP: 240
Title: VET Custodian Contract Standard
```

File: vips/VIP-VET-Custodian.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+Description: A standard for smart contracts to store VET and distribute VTHO back to users fairly and automatically.
+Author: Xiqing Chu <xiqing.chu@vechain.org>
+Discussion: https://vechain.discourse.group/t/add-vip-vet-custodian-contract-standard/44
+Category: Application
+Status: Idea
```
Comment (NecoSherry) [inline vips/VIP-VET-Custodian.md:7]:
```suggestion
Status: Draft
```

**Comment by libotony** [comment]
@NecoSherry, I've made the necessary modifications to the files, including the VIP number. They are now ready for merging.

**Review by sebaseek — APPROVED** [review]

**Pull request merged**